### ISSUE TITLE
[2.7] Need Updated CSI-Proxy to Address Windows Volume Mount Issue

### ISSUE THREAD

**Issue by HarrisonWAffel** [rancher/rancher#44789]
Backport Issue for https://github.com/rancher/rancher/issues/42298

**Comment by HarrisonWAffel** [comment]
The PR description contains details on how this can be tested

**Comment by markusewalker** [comment]
Validated that this is addressed on `v2.7-head` https://github.com/rancher/rancher/commit/bd0bb772b60a447d352789b3d7c4368ffe50eecf.

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/bd0bb772b60a447d352789b3d7c4368ffe50eecf

**TEST RESULT**
|#|Scenario|Test Case|Result
|---|---|---|---|
|1|CSI proxy v1.1.3 establishes multiple connections to smb share|Create a Windows RKE2 cluster, install csi driver smb chart, create a SMB server and validate that it is working|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. As a standard user, provisioned a standard RKE2 cluster ( 1 etcd, 1 cp, 1 Linux worker, 2 Windows worker).
2. Created new Helm repo using `https://raw.githubusercontent.com/kubernetes-csi/csi-driver-smb/master/charts`.
3. Installed `csi-driver-smb` chart.
4. Followed the following guide on creating an SMB server for testing.
5. Exec'ed into one of the deployments and validated `data.txt` is available and is not empty:
```
C:\mnt\smb>dir
 Volume in drive C has no label.
 Volume Serial Number is 180F-1FD9

 Directory of C:\mnt\smb

03/19/2024  07:46 PM    <DIR>          .
03/19/2024  07:46 PM    <DIR>          ..
03/19/2024  09:11 PM           104,874 data.txt
               1 File(s)        104,874 bytes
               2 Dir(s)  92,059,508,736 bytes free
```
